File: mergeTwoSortedArrays.py
#Time:O(n+m)
#Space:O(n+m)
def merge_arrays(arr1, arr2):
    mergedArr = []
    i = 0
    j = 0
    while(i < len(arr1) or j < len(arr2)):
        valueToPush = 0
        if i < len(arr1) and j < len(arr2):
            if arr1[i] < arr2[j]:
                valueToPush = arr1[i]
                i+=1
            else:
                valueToPush = arr2[j]
                j+=1
        else:
            valueToPush = arr1[i] if i < len(arr1) else arr2[j]
            i+=1
            j+=1     
        mergedArr.append(valueToPush)
    return mergedArr
        

File: mergeTwoSortedArrays.cpp
//Time:O(n+m)
//Space:O(n+m)
vector<int> merge_arrays(vector<int> arr1, vector<int> arr2) {
    int indexOne = 0,indexTwo = 0;
    vector<int> mergedArray;
    while(indexOne < (int)arr1.size() || indexTwo < (int)arr2.size())
    {
        int valueToPush = 0;
        if(indexOne < (int)arr1.size() && indexTwo < (int)arr2.size())
        {
            valueToPush = arr1[indexOne] < arr2[indexTwo] ? arr1[indexOne++] : arr2[indexTwo++];
        }
        else
        {
            valueToPush = indexOne < (int)arr1.size() ? arr1[indexOne++] : arr2[indexTwo++];
        }
        mergedArray.push_back(valueToPush);
    }
    return mergedArray;
}